### PR TITLE
allow to disable/enable phases in nixos-anywhere

### PR DESCRIPTION
**Pull request by Mic92**
Instead of having flags like --no-reboot or --stop-after-disko, we now can specify all phases to run.

### CODE REVIEW

File: src/nixos-anywhere.sh [hunk 1]
```diff
@@ -151,11 +159,33 @@ while [[ $# -gt 0 ]]; do
...
     shift
     ;;
+  --phases)
+    phases[kexec]=0
+    phases[disko]=0
+    phases[install]=0
+    phases[reboot]=0
+    IFS=, read -r -a phase_list <<<"$2"
+    for phase in "${phase_list[@]}"; do
+      if [[ ${phases[$phase]:-unset} == unset ]]; then
```
Comment (phaer) [inline src/nixos-anywhere.sh:169]: Could/should have some quotes here, i think?

File: src/nixos-anywhere.sh [hunk 2]
```diff
@@ -364,132 +393,118 @@ if [[ ${is_os-n} != "Linux" ]]; then
...
-  ssh_connection="root@${ssh_host}"
-fi
-
-for path in "${!disk_encryption_keys[@]}"; do
-  step "Uploading ${disk_encryption_keys[$path]} to $path"
-  ssh_ "umask 077; cat > $path" <"${disk_encryption_keys[$path]}"
-done
-
-if [[ ${build_on_remote-n} == "y" ]]; then
-  pubkey=$(ssh-keyscan -p "$ssh_port" -t ed25519 "$ssh_host" 2>/dev/null || {
```
Comment (phaer) [inline src/nixos-anywhere.sh:441]: This (and ssh_port) where removed without replacement, as pubkey was unused anyway. Am I reading this correctly?

**Review by phaer — APPROVED** [review]
Code looks good to me, haven't tested this (yet) :)

File: src/nixos-anywhere.sh (hunk 2 above)
Comment (Mic92) [reply src/nixos-anywhere.sh:441]: Yes. pubkey wasn't used anywhere.

File: src/nixos-anywhere.sh (hunk 1 above)
Comment (Mic92) [reply src/nixos-anywhere.sh:169]: shellcheck disagrees. And the content is also not dynamic.

**Comment by Mic92** [comment]
@mergify queue

**Pull request merged**